token-lending/cli/src/main.rs

```diff
@@ -0,0 +1,700 @@
+        config,
+        lending_market_balance + fee_calculator.calculate_fee(&transaction.message()),
+    )?;
+    transaction.sign(
+        &vec![config.fee_payer.as_ref(), &lending_market_keypair],
+        recent_blockhash,
+    );
+    send_transaction(&config, transaction)?;
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+fn command_add_reserve(
+    config: &Config,
+    ui_amount: f64,
+    reserve_config: ReserveConfig,
+    source_liquidity_pubkey: Pubkey,
+    lending_market_pubkey: Pubkey,
+    lending_market_owner_keypair: Keypair,
+    pyth_product_pubkey: Pubkey,
+    pyth_price_pubkey: Pubkey,
+) -> CommandResult {
+    let source_liquidity_account = config.rpc_client.get_account(&source_liquidity_pubkey)?;
+    let source_liquidity = Token::unpack_from_slice(source_liquidity_account.data.borrow())?;
+
+    let source_liquidity_mint_account = config.rpc_client.get_account(&source_liquidity.mint)?;
+    let source_liquidity_mint =
+        Mint::unpack_from_slice(source_liquidity_mint_account.data.borrow())?;
+    let liquidity_amount = ui_amount_to_amount(ui_amount, source_liquidity_mint.decimals);
+
+    let reserve_keypair = Keypair::new();
+    let collateral_mint_keypair = Keypair::new();
+    let collateral_supply_keypair = Keypair::new();
+    let liquidity_supply_keypair = Keypair::new();
+    let liquidity_fee_receiver_keypair = Keypair::new();
+    let liquidity_host_keypair = Keypair::new();
+    let user_collateral_keypair = Keypair::new();
+    let user_transfer_authority_keypair = Keypair::new();
+
+    println!("Adding reserve {}", reserve_keypair.pubkey());
+    if config.verbose {
+        println!(
+            "Adding collateral mint {}",
+            collateral_mint_keypair.pubkey()
+        );
+        println!(
+            "Adding collateral supply {}",
+            collateral_supply_keypair.pubkey()
+        );
+        println!(
+            "Adding liquidity supply {}",
+            liquidity_supply_keypair.pubkey()
+        );
+        println!(
+            "Adding liquidity fee receiver {}",
+            liquidity_fee_receiver_keypair.pubkey()
+        );
+        println!("Adding liquidity host {}", liquidity_host_keypair.pubkey());
+        println!(
+            "Adding user collateral {}",
+            user_collateral_keypair.pubkey()
+        );
+        println!(
+            "Adding user transfer authority {}",
+            user_transfer_authority_keypair.pubkey()
+        );
+    }
+
+    let reserve_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Reserve::LEN)?;
+    let collateral_mint_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Mint::LEN)?;
+    let collateral_supply_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let user_collateral_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
```

Comment:
no need to refetch here, the value is the same as above

Reply:
Done.